lib/api/api_auth.go

```diff
@@ -182,11 +190,14 @@ func (m *basicAuthAndSessionMiddleware) handleLogout(w http.ResponseWriter, r *h
 }
 
 func auth(username string, password string, guiCfg config.GUIConfiguration, ldapCfg config.LDAPConfiguration) bool {
-	if guiCfg.AuthMode == config.AuthModeLDAP {
-		return authLDAP(username, password, ldapCfg)
-	} else {
-		return authStatic(username, password, guiCfg)
+	if guiCfg.IsPasswordAuthEnabled() {
```

Comment:
I objected to this approach (having "grant access" instead of "deny access" as the default outcome) in https://github.com/syncthing/syncthing/pull/8757#discussion_r1090902419 and got agreement from @imsodin then, but if you still prefer an early return I'll change to that. Please confirm if you'd like me to change this to an early return, if not I'll leave this as is.